tools/src/main/scala/caliban/tools/RemoteSchema.scala

```diff
@@ -0,0 +1,313 @@
+
+      case Type.NamedType(name, nonNull) =>
+        if (nonNull)
+          __Type(
+            kind = __TypeKind.NON_NULL,
+            ofType = Some(
+              toType(name, definitions)
+            )
+          )
+        else
+          toType(name, definitions)
+    }
+  }
+
+  private def toType(
+    name: String,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ) =
+    definitions.find(_.name == name) match {
+      case Some(value) =>
+        value match {
+          case e: EnumTypeDefinition        => toEnumType(e)
+          case s: ScalarTypeDefinition      => toScalar(s)
+          case u: UnionTypeDefinition       => toUnionType(u, definitions)
+          case o: ObjectTypeDefinition      => toObjectType(o, definitions)
+          case i: InputObjectTypeDefinition =>
+            toInputObjectType(i, definitions)
+          case i: InterfaceTypeDefinition   =>
+            toInterfaceType(i, definitions)
+        }
+      case None        => __Type(kind = __TypeKind.SCALAR, name = Some(name))
+    }
+
+  private def toDirectives(directives: List[Directive]): Option[List[Directive]] =
+    if (directives.length > 0) Some(directives.filter(_.name != "deprecated"))
+    else None
+
+  private def toInterfaces(
+    interfaces: List[Type.NamedType],
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ): () => Some[List[__Type]] = { () =>
+    Some(
+      interfaces
+        .map(t => toType(t.name, definitions))
+    )
+  }
+
+  private def toInterfaceType(
+    definition: Definition.TypeSystemDefinition.TypeDefinition.InterfaceTypeDefinition,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ): __Type =
+    __Type(
+      kind = __TypeKind.INTERFACE,
+      name = Some(definition.name),
+      description = definition.description,
+      fields = (args: __DeprecatedArgs) =>
+        if (definition.fields.length > 0)
+          Some(
+            definition.fields
+              .map(t => toField(t, definitions))
+              .filter(filterDeprecated(_, args))
+          )
+        else None,
+      directives = toDirectives(definition.directives)
+    )
+
+  private def toInputValue(
+    definition: Definition.TypeSystemDefinition.TypeDefinition.InputValueDefinition,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ): __InputValue =
+    __InputValue(
+      name = definition.name,
+      description = definition.description,
+      `type` = toType(definition.ofType, definitions),
+      defaultValue = definition.defaultValue.map(_.toString),
+      directives = toDirectives(definition.directives)
+    )
+
+  private def toEnumType(
+    definition: Definition.TypeSystemDefinition.TypeDefinition.EnumTypeDefinition
+  ): __Type =
+    __Type(
+      kind = __TypeKind.ENUM,
+      name = Some(definition.name),
+      enumValues = (args: __DeprecatedArgs) =>
+        if (definition.enumValuesDefinition.length > 0)
```

Comment:
I'd generally prefer to use `nonEmpty` for all these `> 0` checks

Reply:
👍